### ISSUE TITLE
Datagrid page change doesn't trigger resize

### ISSUE THREAD

**Issue by nmussy** [vmware/clarity#2956]
## Describe the bug

Chaning the page in a paginated Datagrid doesn't trigger a resize of the Datagrid, and its content has to be scrolled by the user.

## How to reproduce

[Stackblitz example](https://stackblitz.com/edit/clarity-datagrid-pagination-resize?file=src%2Fapp%2Fapp.component.html)

1.  Go to the next page
2.  The `.datagrid-table` isn't resized
3. Change the pagination to 5 items a page
3. The `.datagrid-table` is resized
4.  Go to the next page
5. The `.datagrid-table` isn't resized

## Expected behavior

`(clrDgPageChange)` should trigger a Datagrid resize. Changing `[clrDgPageSize]` forces a resize, but is lost again when changing the page.

## Versions

**App**

* Angular: 7
* Clarity: 1.0.2

**Device:**

* Type: Desktop
* OS: Windows
* Browser: Chrome
* Version: 71

**Comment by hippee-lee** [comment]
This is the behavior I would expect. Here is why: When we change `clrDgPageSize` we need to update the height of the datagrid based on the new number of rows to display. However, when changing the page we cannot know what content will be put into the datagrid so we don't update the height to keep a consistent datagrid height for when the last page is reached. We used to have the datagrid automatically adapt its height to the content projected in the rows but this meant when a user reached the last page if it only had one row there could be a pretty big jump as the datagrid height shrunk do adapt to that change. 

To recap, when a datagrid is set up with pagination we calculate its height based on the rows given and we don't recalculate the height unless the number of rows displayed changes so that the datagrid height doesn't jump up and down as either content in subsequent pages changes or they reach the last page and there is only one row on that page. 

I'm closing this because its expected behavior. Please reopen with additional details if I misunderstand the report.

**Comment by Jinnie** [comment]
Hi, @nmussy
Additionally to what @hippee-lee said, even only growing the size would be a problem, as it will reposition the footer with the page navs. If the footer is jumping up and down, browsing through pages will be a bad experience, as the user will need to seek the next-page button after every click. If you expect to have a lot of data in some cell, you might consider the expandable row feature.

**Comment by nmussy** [comment]
Fair enough, thank you both for your time.
